gateway/ln-gateway/src/state_machine/mod.rs

```diff
@@ -351,7 +351,7 @@ impl GatewayClientModule {
         time_to_live: Duration,
         fees: RoutingFees,
         lightning_context: LightningContext,
-    ) -> anyhow::Result<()> {
```

Comment:
It's weird that it will fail without notifying the caller. At least you can change this function name to `try_register_with_federation` to let this behavior be clear.

Reply:
Done